src/test/architecture/test_architecture.py

```diff
@@ -312,3 +312,64 @@ def test_verify_callback_uniqueness(self, mocker, caplog):
         caplog.clear()
         Architecture._verify([node_mock])
         assert len(caplog.record_tuples) == 1
+
+    def test_assign_function(self, mocker):
+        architecture_text = """
+named_paths: []
+executors: []
+nodes:
+- node_name: /pong_node
+  callbacks:
+    - callback_name: subscription_callback_0
+      callback_type: subscription_callback
+      topic_name: /pong
+      symbol: sub_symbol
+    - callback_name: timer_callback_1
+      callback_type: timer_callback
+      period_ns: 1
+      symbol: timer_symbol
+  subscribes:
+    - topic_name: /pong
+      callback_name: subscription_callback_0
+"""
+
+        mocker.patch('builtins.open', mocker.mock_open(read_data=architecture_text))
+        arch = Architecture('yaml', 'architecture.yaml')
+        node = arch.get_node("/pong_node")
+
+        architecture_text_expected = """
+named_paths: []
+executors: []
+nodes:
+- node_name: /pong_node
+  callbacks:
+    - callback_name: subscription_callback_0
+      callback_type: subscription_callback
+      topic_name: /pong
+      symbol: sub_symbol
+    - callback_name: timer_callback_1
+      callback_type: timer_callback
+      period_ns: 1
+      symbol: timer_symbol
+  variable_passings:
+    - callback_name_write: subscription_callback_0
+      callback_name_read: timer_callback_1
+  publishes:
+    - topic_name: /ping
+      callback_names:
+        - timer_callback_1
+  subscribes:
+    - topic_name: /pong
+      callback_name: subscription_callback_0
+  message_contexts:
+    - context_type: callback_chain
+      subscription_topic_name: /pong
+      publisher_topic_name: /ping
+"""
+
+        mocker.patch('builtins.open', mocker.mock_open(read_data=architecture_text_expected))
+        arch = Architecture('yaml', 'architecture.yaml')
+        node_expected = arch.get_node("/pong_node")
+
+        assert len(node.variable_passings) == len(node_expected.variable_passings)
```

Comment:
VariableObjects can compare themselves.

```suggestion
        assert node.variable_passings == node_expected.variable_passings
        assert node.publishers == node_expected.publishers
```

Reply:
Fixed it in [50b4698](https://github.com/tier4/CARET_analyze/pull/196/commits/50b4698dffa5b3c7368e101c3185d1f8b5e47ef0)